scan/context.go

```diff
@@ -100,7 +100,8 @@ func (s *Scanner) getContextFromReader(r io.Reader) (err error) {
 	buf := bufio.NewReader(r)
 	var magic []byte
 
-	magic, err = buf.Peek(archiveHeaderSize)
+	// note: (sam) read 2048 magic bytes to accomodate Extended Pax headers.
+	magic, err = buf.Peek(archiveHeaderSize * 4)
```

Comment:
dump question, if the tarball length is less than 2K, will Peek return EOF?

Reply:
Yes, Peek return "EOF" which is handled below.